modules/javafx.base/src/main/java/javafx/beans/binding/ObjectBinding.java

```diff
@@ -180,7 +180,10 @@ public final void invalidate() {
             valid = false;
             onInvalidating();
             ExpressionHelper.fireValueChangedEvent(helper);
-            value = null;  // clear cached value to avoid hard reference to stale data
+
+            if (!valid) {  // if still invalid after calling listeners...
```

Comment:
On a personal note: I think implementation comments are best presented as a block, and not describing individual lines of code. That makes it a bit easier for future authors to work with the code. But that's just my two cents 😊